### PR TITLE
Docs: Update Iceberg Hive documentation - 1.0.x (#6337)

### PR DESCRIPTION
**Pull request by InvisibleProgrammer**
Docs: Update Iceberg Hive documentation - 1.0.x 

(cherry picked from commit 5b1505378414295476a73e07aca65fdb1a29da7d)

### CODE REVIEW

**Comment by InvisibleProgrammer** [comment]
1.0.x port of https://github.com/apache/iceberg/pull/6337

**Comment by ajantha-bhat** [comment]
There won't be any release from the 1.0.x branch. So, this backport is not really required.   
Keeping the fix in the master branch is enough. The next 1.2.0 release will be from the master branch.

**Comment by InvisibleProgrammer** [comment]
@ajantha-bhat, I understand that there will be no release on a released version of Iceberg. Also, I want to understand why it is not recommended to update the documentation. Does the documentation have a release flow separated from the Iceberg code or it can be released only as a whole package? And I wonder, what is the reason behind freezing the documentation? 
Does it mean that for the future release, we must document everything upfront? 

Thank you,
Zsolt

**Comment by pvary** [comment]
@ajantha-bhat: 2 questions:
- Does this mean, that we will never release an 1.0.1? Even if we find a `log4j` level serious issue?
- Does this mean, that if we want to update the current documentation (both for current and older releases) we need to create a pull request for https://github.com/apache/iceberg-docs?

Thanks,
Peter

**Comment by ajantha-bhat** [comment]
> Does this mean, that we will never release an 1.0.1? Even if we find a log4j level serious issue?

I have not seen this case, might just release a new version of the last release. (ex: 1.1.1), but may not be for 1.0.X or older. 
@Fokko or other previous release owners can answer this may be. 

> Does this mean, that if we want to update the current documentation (both for current and older releases) we need to create a pull request for https://github.com/apache/iceberg-docs?

There are two kinds of docs, one that is in `iceberg` repo and one that is in `iceberg-docs `repo. 
`iceberg-docs` repo has some site-related extra docs and it also refers to `iceberg` repo docs. The release owner creates branches in `iceberg-docs ` repo. So we need to raise PR wherever our `.md` file is present related to that change.

**Comment by InvisibleProgrammer** [comment]
I think I've found the proper way to update the documentation for older versions. What do you think, is that the correct way? 

- 0.14.1: https://github.com/apache/iceberg-docs/pull/188
- 1.0.0: https://github.com/apache/iceberg-docs/pull/189
- 1.1.0: https://github.com/apache/iceberg-docs/pull/190

Thank you, 
Zsolt

**Comment by InvisibleProgrammer** [comment]
Hey there!

With the help of @pvary , all updates for the older releases are merged. 

I wonder, do you know what is the process of releasing the merged changes?

File: docs/hive.md [hunk 1]
```diff
@@ -38,6 +38,16 @@ Iceberg compatibility with Hive 2.x and Hive 3.1.2/3 supports the following feat
 DML operations work only with MapReduce execution engine.
 {{< /hint >}}
 
+With Hive version 4.0.0-alpha-2 and above,
```
Comment (rdblue) [inline docs/hive.md:41]: Why does this refer to an alpha release? I think we should only reference official Hive releases, and probably not alpha ones.

File: docs/hive.md [hunk 2]
```diff
@@ -244,7 +254,7 @@ The result is:
 | j                                  | IDENTITY       | NULL
 
 You can create Iceberg partitions using the following Iceberg partition specification syntax
-(supported only in Hive 4.0.0-alpha-1):
+(supported only from Hive 4.0.0-alpha-1):
```
Comment (rdblue) [inline docs/hive.md:257]: How does this create partitions? Is this trying to say you can alter partition fields?

File: docs/hive.md [hunk 3]
```diff
@@ -433,6 +449,15 @@ Tables can be dropped using the `DROP TABLE` command:
 DROP TABLE [IF EXISTS] table_name [PURGE];
 ```
 
+### METADATA LOCATION
```
Comment (rdblue) [inline docs/hive.md:452]: What is the value of this? Is it used for something?

File: docs/hive.md [hunk 4]
```diff
@@ -509,7 +534,15 @@ SELECT * FROM table_a FOR SYSTEM_TIME AS OF '2021-08-09 10:35:57';
 SELECT * FROM table_a FOR SYSTEM_VERSION AS OF 1234567;
 ```
 
-## Type compatibility
+You can expire snapshots of an Iceberg table using an ALTER TABLE query from Hive. You should periodically expire snapshots to delete data files that is no longer needed, and reduce the size of table metadata.
```
Comment (rdblue) [inline docs/hive.md:537]: I think this needs a section heading.

File: docs/hive.md (hunk 1 above)
Comment (pvary) [reply docs/hive.md:41]: Hive is pushing for the final release. There were no new releases for a few years now, but the community started to finalize one. And one of the main feature there is the native Iceberg support which is not there with older releases. We need to notify the commit about the new features to gather feedback and fix issues for the final release.

File: docs/hive.md (hunk 2 above)
Comment (InvisibleProgrammer) [reply docs/hive.md:257]: I wouldn't say that. It is about creating a table, not altering one.

**Pull request closed without merging**